x-pack/plugins/security_solution/public/alerts/components/value_lists_management_modal/modal.tsx

```diff
@@ -0,0 +1,161 @@
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useCallback, useEffect, useState } from 'react';
+import {
+  EuiButton,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiOverlayMask,
+  EuiSpacer,
+} from '@elastic/eui';
+
+import {
+  ListSchema,
+  exportList,
+  useFindLists,
+  useDeleteList,
+  useCursor,
+} from '../../../lists_plugin_deps';
+import { useToasts, useKibana } from '../../../common/lib/kibana';
+import { GenericDownloader } from '../../../common/components/generic_downloader';
+import * as i18n from './translations';
+import { ValueListsTable } from './table';
+import { ValueListsForm } from './form';
+
+interface ValueListsModalProps {
+  onClose: () => void;
+  showModal: boolean;
+}
+
+export const ValueListsModalComponent: React.FC<ValueListsModalProps> = ({
+  onClose,
+  showModal,
+}) => {
+  const [pageIndex, setPageIndex] = useState(0);
+  const [pageSize, setPageSize] = useState(5);
+  const [cursor, setCursor] = useCursor({ pageIndex, pageSize });
+  const { http } = useKibana().services;
+  const { start: findLists, ...lists } = useFindLists();
+  const { start: deleteList } = useDeleteList();
+  const [exportListId, setExportListId] = useState<string>();
+  const toasts = useToasts();
+
+  const fetchLists = useCallback(() => {
+    findLists({ cursor, http, pageIndex: pageIndex + 1, pageSize });
+  }, [cursor, http, findLists, pageIndex, pageSize]);
+
+  const handleDelete = useCallback(
+    async ({ id }: { id: string }) => {
+      await deleteList({ http, id });
+      fetchLists();
+    },
+    [deleteList, http, fetchLists]
+  );
+
+  const handleExport = useCallback(
+    async ({ ids }: { ids: string[] }) =>
+      exportList({ http, listId: ids[0], signal: new AbortController().signal }),
+    [http]
+  );
+  const handleExportClick = useCallback(({ id }: { id: string }) => setExportListId(id), []);
+  const handleExportComplete = useCallback(() => setExportListId(undefined), []);
+
+  const handleTableChange = useCallback(
+    ({ page: { index, size } }: { page: { index: number; size: number } }) => {
+      setPageIndex(index);
+      setPageSize(size);
+    },
+    [setPageIndex, setPageSize]
+  );
+  const handleUploadError = useCallback(
+    (error: unknown) => {
+      if (!String(error).includes('AbortError')) {
+        const reportedError = error instanceof Error ? error : new Error(String(error));
+        toasts.addError(reportedError, { title: i18n.UPLOAD_ERROR });
+      }
+    },
+    [toasts]
+  );
+  const handleUploadSuccess = useCallback(
+    (response: ListSchema) => {
+      toasts.addSuccess({
+        text: i18n.uploadSuccessMessage(response.name),
+        title: i18n.UPLOAD_SUCCESS_TITLE,
+      });
+      fetchLists();
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Is it possible or easy to remove the disabling of the linter rule?

Reply:
The story here is [in the commit that added it](0abbfc422a6356db70cacea2834f08d2cf214420). I agree that this is a smell and I'll try it again from a different angle. If you have suggestions, though, I'd appreciate them! Ditto on `useCursor`.